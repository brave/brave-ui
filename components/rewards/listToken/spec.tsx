/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import ListToken from './index'

describe('ListToken tests', () => {
  const baseComponent = (props?: object) => <ListToken id='list' {...props} />

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#list').length
      expect(assertion).toBe(1)
    })
  })
})
