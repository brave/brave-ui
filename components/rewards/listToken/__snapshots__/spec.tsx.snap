// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListToken tests basic tests matches the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #d0d6dc;
  border-top: none;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-bottom: 8px;
  font-family: Poppins;
}

.c1 {
  font-size: 14px;
  line-height: 2.79;
  color: #4b4c5c;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 60%;
  -ms-flex-preferred-size: 60%;
  flex-basis: 60%;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
  text-align: right;
}

.c3 {
  font-family: Poppins;
  font-weight: 300;
  line-height: 1.4;
  color: #686978;
  font-size: 14px;
  display: inline-block;
}

.c4 {
  color: #686978;
  font-weight: 600;
}

<div
  className="c0"
  id="list"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  >
    <span
      id={undefined}
    >
      <span
        className="c3"
      >
        <span
          className="c4"
        >
          
        </span>
         
        <span>
          tokens
        </span>
      </span>
    </span>
  </div>
</div>
`;
